gen/calc.py

```diff
@@ -419,6 +429,68 @@ def calculate_cluster_docker_registry_enabled(cluster_docker_registry_url):
     return 'false' if cluster_docker_registry_url == '' else 'true'
 
 
+def validate_cosmos_config(cosmos_config):
+    """The schema for this configuration is.
+    {
+      "schema": "http://json-schema.org/draft-04/schema#",
+      "type": "object",
+      "properties": {
+        "staged_package_storage_uri": {
+          "type": "string"
+        },
+        "package_storage_uri": {
+          "type": "string"
+        }
+      }
+    }
+    """
+    config = validate_json_dictionary(cosmos_config)
+    if ('staged_package_storage_uri' in config and
+            'package_storage_uri' not in config):
+        raise AssertionError(
+            '"cosmos_config.stage_package_storage_uri" ({}) specified but '
+            '"cosmos_config.package_storage_uri" was not specified'.format(
+                config['staged_package_storage_uri']
+            )
+        )
+    elif ('staged_package_storage_uri' not in config and
+          'package_storage_uri' in config):
+        raise AssertionError(
+            '"cosmos_config.package_storage_uri" ({}) specified but '
+            '"cosmos_config.stage_package_storage_uri" was not '
+            'specified'.format(
+                config['package_storage_uri']
+            )
+        )
+    elif ('staged_package_storage_uri' in config and
+          'package_storage_uri' in config):
+        # Let's try to parse the string as an URL
+        validate_url(config['staged_package_storage_uri'])
+        validate_url(config['package_storage_uri'])
+
+
+def calculate_cosmos_staged_package_storage_uri_flag(cosmos_config):
+    if 'staged_package_storage_uri' in cosmos_config:
```

Comment:
this is doing string searching rather than dictionary lookup. Can we switch this around to

1. load_json(cosmos_config)
2. check if key exists and if so return
3. else return empty